docs/streams/developer-guide.html

```diff
@@ -163,46 +163,190 @@
 
     <h4><a id="streams_processor_statestore" href="#streams_processor_statestore">State Stores</a></h4>
 
-    <p>
-        Note that the <code>Processor</code> API is not limited to only accessing the current records as they arrive in the <code>process()</code> method, but can also maintain processing states
-        that keep recently arrived records to use in stateful processing operations such as windowed joins or aggregation.
-        To take advantage of these states, users can define a state store by implementing the <code>StateStore</code> interface (the Kafka Streams library also has a few extended interfaces such as <code>KeyValueStore</code>);
-        in practice, though, users usually do not need to customize such a state store from scratch but can simply use the <code>Stores</code> factory to define a state store by specifying whether it should be persistent, log-backed, etc.
-        In the following example, a persistent key-value store named "Counts" with key type <code>String</code> and value type <code>Long</code> is created.
-    </p>
+<p>
+In order to make state stores fault-tolerant (e.g., to recover from machine crashes) as well as to allow for state store migration without data loss (e.g., to migrate a stateful stream task from one machine to another when elastically adding or removing capacity from your application &lt;streams_developer-guide_execution-scaling&gt;), a state store can be <strong>continuously backed up</strong> to a Kafka topic behind the scenes. We sometimes refer to this topic as the state store's associated <em>changelog topic</em> or simply its <em>changelog</em>. In the case of a machine failure, for example, the state store and thus the application's state can be fully restored from its changelog. You can enable or disable this backup feature &lt;streams_developer-guide_state-store_enable-disable-fault-tolerance&gt; for a state store, and thus its fault tolerance.
```

Comment:
I guess you probably just copied this from cp docs, but could we try and do 1 sentence per line? It will help in the future if/when we make changes.